cosmic_ray/operators/unary_operator_replacement.py

```diff
@@ -27,21 +27,51 @@ def _to_ops(from_op):
             yield to_op
 
 
-class MutateUnaryOperator(Operator):
-    """An operator that modifies unary operators."""
+class ReplaceUnaryOperator(ReplacementOperator):  # pylint: disable=abstract-method
+    """An operator that modifies unary operators.
+
+    This is subclassed for each valid from/to unary operator pair, and each
+    subclass provides `from_op` and `to_op` class-level attributes. Each of
+    these is an `ast.Node` or `None`.
+    """
 
-    def visit_UnaryOp(self, node):  # pylint: disable=invalid-name
-        """
-            http://greentreesnakes.readthedocs.io/en/latest/nodes.html#UnaryOp
-        """
-        return self.visit_mutation_site(
-            node,
-            len(build_mutations([type(node.op)], _to_ops)))
+    def visit_UnaryOp(self, node):  # pylint: disable=invalid-name, missing-docstring
+        if isinstance(node.op, self.from_op):
+            return self.visit_mutation_site(node)
+        return node
 
-    def mutate(self, node, idx):
+    def mutate(self, node, _):
```

Comment:
My IDE lights up like a Christmas tree when arguments are ignored with `_` if you're overriding a method. I work around this by asserting on the argument just inside the method. For example.

    assert idx >= 0